pkg/combustion/rpm.go

```diff
@@ -32,13 +33,25 @@ func configureRPMs(ctx *image.Context) ([]string, error) {
 
 	zap.L().Info("Configuring RPM component...")
 
-	var rpmDir string
+	var localRPMConfig *image.LocalRPMConfig
 	if isComponentConfigured(ctx, userRPMsDir) {
-		rpmDir = generateComponentPath(ctx, userRPMsDir)
+		rpmDir := generateComponentPath(ctx, userRPMsDir)
+		localRPMConfig = &image.LocalRPMConfig{
+			RPMPath: rpmDir,
+		}
+
+		gpgPath := filepath.Join(rpmDir, userGPGsDir)
+		_, err := os.Stat(gpgPath)
```

Comment:
Do we want to check the `NoGPGCheck` flag here? Audit and log the warning if true and fail right away if true but the path exists?

Reply:
Added the following validation:

1. If the `gpg-keys` dir exists, but  `noGPGChecks: true`, we audit and log a warning
2. If the `gpg-keys` dir does not exist, but `noGPGChecks: false`, we fail right away and do not go into any resolution logic